File: Heat/ViewModels/Invoices/AddNewDescriptiveInvoiceRowViewModel.vb
Imports System.ComponentModel.DataAnnotations

Namespace ViewModels.Invoices
    Public Class AddNewDescriptiveInvoiceRowViewModel
        Property InvoiceID As Integer

        <Required> _
        <Display(Name:="Descrizione")> _
        Property RowDescription As String

        <Required> _
        <Display(name:="Quantità")> _
        <Range(0, Integer.MaxValue)> _
        Property Quantity As Single

        <Display(name:="Prezzo")> _
        <DataType(DataType.Currency)> _
        Property UnitPrice As Decimal

        <Display(name:="IVA")> _
        Property VAT As Double

        <Display(name:="Sconto 1")> _
        Property Discount1 As Single

        <Display(name:="Sconto 2")> _
        Property Discount2 As Single

        <Display(name:="Sconto 3")> _
        Property Discount3 As Single


    End Class
End Namespace

File: Heat.ConvertedToC#/ViewModels/Invoices/AddNewDescriptiveInvoiceRowViewModel.cs
using System.ComponentModel.DataAnnotations;

namespace Heat.ViewModels.Invoices
{
    public class AddNewDescriptiveInvoiceRowViewModel
	{
		public int InvoiceID { get; set; }

		[Required()]
		[Display(Name = "Descrizione")]
		[DataType(DataType.MultilineText)]
		public string RowDescription { get; set; }

		[Required()]
		[Display(Name = "Quantità")]
		[Range(0, int.MaxValue)]
		public float Quantity { get; set; }

		[Display(Name = "Prezzo")]
		[DataType(DataType.Currency)]
		public decimal UnitPrice { get; set; }

		[Display(Name = "IVA")]
		public double VAT { get; set; }

		[Display(Name = "Sconto 1")]
		public float Discount1 { get; set; }

		[Display(Name = "Sconto 2")]
		public float Discount2 { get; set; }

		[Display(Name = "Sconto 3")]
		public float Discount3 { get; set; }


	}
}

